### PR TITLE
feat(@ciscospark/i-p-board): allow delete channel from a spark space

### PR DESCRIPTION
**Pull request by tran2**
Mainly remove the link between a whiteboard and a spark space.

### CODE REVIEW

File: packages/node_modules/@ciscospark/internal-plugin-board/src/board.js [hunk 1]
```diff
@@ -128,6 +128,37 @@ const Board = SparkPlugin.extend({
...
+  deleteChannel(conversation, channel) {
+    return this.spark.request({
+      method: `PUT`,
+      uri: `${channel.aclUrl}/links`,
+      body: this._prepareDeleteChannel(conversation, channel)
+    })
+      .then((res) => res.body);
+  },
+
+  _prepareDeleteChannel(conversation, channel) {
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/internal-plugin-board/src/board.js]: Is this something likely to be used anywhere other than deleteChannel? if not, not sure it needs to be its own method.

File: packages/node_modules/@ciscospark/internal-plugin-board/src/board.js [hunk 2]
```diff
@@ -128,6 +128,37 @@ const Board = SparkPlugin.extend({
...
+    // remove the ACL link between conversation and board
+    // remove conversation auth from board KRO in kms message
+    return {
+      aclLinkType: `INCOMING`,
+      linkedAcl: conversation.aclUrl,
+      kmsMessage: {
+        method: `delete`,
+        uri: `${channel.kmsResourceUrl}/authorizations?${querystring.stringify({authId: conversation.kmsResourceObjectUrl})}`
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/internal-plugin-board/src/board.js]: note that request accepts a `qs` property and takes care of stringification if you give it a hash automatically

Comment (tran2) [reply packages/node_modules/@ciscospark/internal-plugin-board/src/board.js]: this uri is for the kmsMessage though, would request take care of that?

Comment (ianwremmel) [reply packages/node_modules/@ciscospark/internal-plugin-board/src/board.js]: yep, keep 
```js
uri: `${channel.kmsResourceUrl}/authorizations`
```
then just put the querystring params in the `qs` hash.

**Pull request merged**